consensus/src/quorum_store/batch_generator.rs

```diff
@@ -50,9 +65,9 @@ pub struct BatchGenerator {
     batch_writer: Arc<dyn BatchWriter>,
     config: QuorumStoreConfig,
     mempool_proxy: MempoolProxy,
-    batches_in_progress: HashMap<BatchId, Vec<TransactionSummary>>,
+    batches_in_progress: HashMap<(PeerId, BatchId), BatchInProgress>,
```

Comment:
Slightly orthogonal to this PR, but are there protection against malicious validator DDOSing the system and filling up the quorum store with its own batches?